app/controllers/api/v2/courses_controller.rb

```diff
@@ -25,6 +25,12 @@ def sync_with_search_and_compare
         head response ? :ok : :internal_server_error
       end
 
+      def publish
+        # todo: set running
+        # todo: set enrichment published
+        sync_with_search_and_compare
```

Comment:
This smells, to me, is the intention to change this when the other logic is applied?

I don't think I've seen an action call another action like this. Probably because each action needs to build it's own response to an action, otherwise it should be an alias.

Reply:
I'm not sure what an alias would look like, could you link to an example for my edification?

Intention is to try and get a PR approved :laughing: - but seriously, this is a bit of a straw-man, not sure what the best way of DRY'ing this would be or whether we should DRY it at all. We could duplicate the logic from the sync method.

Reply:
I would duplicate the logic. One way to do this is to have a service that encapsulates it (e.g. `CourseSyncWithSearchAndCompareService`) independent of anything else like the params coming, how to generate a response, etc. Even in this case where the logic simply just defers to `ManageCoursesAPI::Request.sync_course_with_search_and_compare`, it would still be better, notionally, as a separate service.

Reply:
done the duplication for now as we discussed